### DISCUSSION TITLE
Downloading workshop mods when joining a server no longer works

### DISCUSSION THREAD

**Discussion by SquidingTin** [FakeFishGames/Barotrauma#13695] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The game kicks you to the main menu when you try to download mods from workshop. This appears to have only started happening after today's update. The Epic Games patch update.

I can get multiple people to verify that this isn't an isolated issue either.

### Reproduction steps

1. Find a multiplayer server running mods you aren't currently subscribed to.
2. Press "Subscribe to all on workshop"
3. get kicked to main menu

https://github.com/FakeFishGames/Barotrauma/assets/33539475/8cbbece8-cb4c-4e43-a653-eed2f6babfa9

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.3.0.0/v1.3.0.1

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
WARNING: Tried to restore enabled packages multiple times/without performing a backup
```

**Comment by GoodGuyJimmy** [comment]
Can confirm it also happens to me. People can no longer join my modded server unless they manually download all the mods from the workshop before joining.

**Comment by NotWendy** [comment]
I also had some serious trouble downloading mods from the server I regularly play on. After pressing "Yes" in that menu to download mods from the server itself, the first mod would get stuck at 0B/0B. 7 times, until it finally worked. No idea if this was caused by the update in some way or not. Happened before the update too, but not in such amounts.
If anyone else has encountered that, let me know.

**Reply by GoodGuyJimmy** [reply]
2 of my friends had that same problem when trying to join my server a few days ago the day the update released. We haven't had that problem before the update except for in some rare occasions that were usually fixed by either just joining again or restarting the game. So I guess that's also a problem with the update since they were only able to join after manually downloading the new mods.

**Comment by Regalis11** [comment]
Thank you for the report! It seems like some of the refactoring we did in the latest update broke this. I've created [a bug ticket of the issue](https://github.com/FakeFishGames/Barotrauma/issues/13705), we'll push a hotfix as soon as we can, hopefully early next week. Sorry for the inconvenience!

**Comment by Regalis11** [comment]
The fix should now be live in the public version. Again, sorry for the inconvenience!
